test/reusing-processes/index.js

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const assert = require('assert');
+const { resolve } = require('path');
+const { exec } = require('child_process');
+const libExecutable = resolve(__dirname, '../../dist/bin/cli.js');
+
+exec(`${libExecutable} --max-parallel 1 test/reusing-processes/index.spec.js test/reusing-processes/index.2.spec.js`, {
+    cwd: resolve(__dirname, '../../'),
+}, (err, stdout) => {
+    if (err) {
+        console.error(err);
+        process.exit(1);
+    }
+
+    const [pid1, pid2] = stdout.split(/\n/g).filter(line => /^pid/.test(line));
+
+    if (pid1 && pid2) assert.strictEqual(pid1, pid2);
```

Comment:
This looks a bit fragile: if one of the tests doesn't output anything this test will still pass. Do we actually need this check for `pid1` and `pid2` existence? Isn't it safe to just remove this condition check and run assertion check instead?

Reply:
Sometimes they don't both come out...

Reply:
Well, this means that a test is flaky and it should be investigated why it's so. Let's investigate that or remove the test altogether.

Reply:
I'd still posit that the test has value. It works 90% of the time and could catch a bug in that 90%.

Reply:
I have a very strong opinion on that. Saying that in 10% of tests one of `pid1` or `pid2` are empty means that the test hasn't been executed at all because this means that the main process has finished its execution but one of the files which were supposed to get executed didn't do it. This looks like a major bug and I'd spend more time into that. You can skip that, but I will do that later on.

Reply:
Oh btw I think I know why this can be flaky thing: standard streams are considered to be asynchronous (i.e. console.log is using process.stdout.write which is asynchronous) and this is why listening to their output can be tricky. What you can do is to use a synchronous function to write the process ID for instance `fs.writeFileSync`. This would guarantee that your PID is saved before the test file has finished executing.